### DISCUSSION TITLE
Passing step number to agent step method

### DISCUSSION THREAD

**Discussion by erissatallan** [mesa/mesa#1545] [Q&A] [answered]
My AgentModel class has methods that increment the likelihood that an agent will maintain ownership of an asset in response to incentive given to the agent. One of these methods is
```
def sigmoidal_incentive_mechanism(self, x):
        self.maintain_probability = 0.1 + (0.9 / 1 + math.exp(-1 * incentive_pool * (self.token_wealth / economy_token_wealth) * x))
```
One that employs the behavior of the sigmoid function to start this probability at 0.55 and always remain between 0 and 1.
x is meant to capture what number of step the model is being run for, following from this
```
economyModel = EconomyModel(100, 10, 10)
for run_i in range(10):
    economyModel.step(run_i)
```
where my EconomyModel class' step method is
```
def step(self, run_i):
    self.datacollector.collect(self)
    self.schedule.step()
```
I want to figure out where to pass `run_i` here so that it is passed to the AgentModel class' `step()` which in turn passes it to the methods updating the aforementioned probability
```
def step(self, run_i):
    self.sigmoidal_incentive_mechanism(run_i)
```
How would one go about that?

**Comment by rht** [comment, marked as answer]
You don't have to pass the `run_i`. The scheduler keeps track of the step count (see https://github.com/projectmesa/mesa/blob/9bc7b1a3a998c68dec2b85d259baedbf3e6bce02/mesa/time.py#L83). You just need to access this scheduler attribute from your agent.

**Reply by erissatallan** [reply]
Thank you. Could you suggest how to pass EconomyModel attributes to AgentModel during every step? I have some attributes that, unlike the step number, have not been provided for by mesa. So far I haven't thought up any way to do it other than schedule's  `step()` since it is the only method that calls `step()` in AgentModel.

**Reply by tpike3** [reply]
If I am understanding correctly, when you set up your AgentModel Class you should also pass in your model. See example below form the mesa tutorial

```
class MoneyAgent(mesa.Agent):
    """An agent with fixed initial wealth."""

    def __init__(self, unique_id, model):
        super().__init__(unique_id, model)
        self.wealth = 1
``` 

Your agent model can then reference any attribute of your EconomyModel by referencing `self.model.<model attribute to reference>`

**Reply by erissatallan** [reply]
This works. Thank you.
